glossary.md

```diff
@@ -0,0 +1,172 @@
+_Module 6_
+
+**API** - "Application Programming Interface", a defined interface between two computer systems. We use a few web APIs using JSON over HTTP during workshops and the project exercise.  
+_Module 2_
+
+**Authentication** - The process of verifying the identity of a user. Sometimes referred to as AuthN.  
+_Module 10_
+
+**Authorisation** - Determining what resources a user has access to. Sometimes referred to as AuthZ.  
+_Module 10_
+
+**Blameless culture** - A workplace culture that encourages looking for potential system improvements after incidents occur rather than assigning blame to individuals.
+One of the core principals of DevOps culture.  
+_Module 6_
+
+**Branching by abstraction** - A technique for avoiding long lived branches for large feature changes by introducing an abstraction that can switch between two underlying implementations, often using a feature toggle.
+See <https://martinfowler.com/bliki/BranchByAbstraction.html>.  
+_Module 2_
+
+**Deming cycle**/plan-do-check-act - A process/methodology for making improvements to a system or organisation. See <https://en.wikipedia.org/wiki/PDCA>.  
+_Module 6_
+
+**Distributed Source Control** - A source control/version control system where each user has a full copy of the repository. Git is a good example of this.  
+_Module 1_
+
+**Continuous Integration/CI** - The practice of merging all developers' working copies to a shared main branch regularly.
+A Continuous Integration pipeline assists with this by checking the state of repository branches, that might include running tests, static analysis tools and building the application.  
+_Module 7_
+
+**Continuous Delivery/CD** - The ability to deploy software at any time using an automated deployment pipeline.
+The pipeline may still have steps that have to be manually approved.  
+_Module 8_
+
+**Continuous Deployment** - An extension of Continuous Delivery where each viable commit to a main branch is automatically deployed through to production with no manual intervention.  
+_Module 8_
+
+**CPD** - "Continued Professional Development", continually improving your own knowledge and skills.  
+_Module 6_
+
+**Dependency Checking** - Automated analysis of a systems dependencies to ensure they do not have any known vulnerabilities.  
+_Module 10_
+
+**Feature toggle** - A flag used to alter the behaviour of a system, for example to turn a new feature on and off.
+Using feature toggles allows short lived branches to be used for large features that need to be turned on all at once.  
+_Module 2_
+
+**Functional/non-functional requirements** - Functional requirements pertain to features of an application.
+Non-functional requirements (NFRs) describe other attributes of a system, such as performance, accessibility and security.
+NFRs are often not explicitly written down on tickets, but are nonetheless critical to building a working application.  
+_Module 6_
```

Comment:
Our reading material doesn't dive into the definition in much detail, would it be worth linking a blog post for this? Maybe <https://www.perforce.com/blog/alm/what-are-non-functional-requirements-examples>?